File: src/main/java/com/heybys/oddments/fooddelivery/controller/shop/ShopResponse.java
package com.heybys.oddments.fooddelivery.controller.shop;

import java.math.BigDecimal;

import com.heybys.oddments.fooddelivery.domain.shop.Shop;

import lombok.Getter;

@Getter
public class ShopResponse {

    private Long shopId;
    private String shopName;
    private BigDecimal shopMinOrderPrice;

    public static ShopResponse of(Shop shop) {
        ShopResponse response = new ShopResponse();
        response.shopId = shop.getId().longValue();
        response.shopName = shop.getName();
        response.shopMinOrderPrice = shop.getMinOrderPrice().getAmount();
        return response;
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/controller/shop/ShopResponse.kt
package com.heybys.oddments.fooddelivery.controller.shop

import com.heybys.oddments.fooddelivery.domain.shop.Shop
import java.math.BigDecimal

data class ShopResponse(val shopId: Long, val shopName: String, val shopMinOrderPrice: BigDecimal) {

    companion object {
        @JvmStatic
        fun of(shop: Shop): ShopResponse {
            return ShopResponse(
                shopId = shop.getId()!!.longValue(),
                shopName = shop.name!!,
                shopMinOrderPrice = shop.minOrderPrice!!.amount,
            )
        }
    }
}
